src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -195,11 +200,19 @@ trait FastSync {
       case SyncRequestHandler.Done =>
         context unwatch sender()
         assignedHandlers -= sender()
+        requestedMptNodes = requestedMptNodes - sender()
+        requestedNonMptNodes = requestedNonMptNodes - sender()
+        requestedBlockBodies = requestedBlockBodies - sender()
+        requestedReceipts = requestedReceipts - sender()
         processSyncing()
 
       case Terminated(ref) if assignedHandlers.contains(ref) =>
         context unwatch ref
         assignedHandlers -= ref
+        requestedMptNodes = requestedMptNodes - ref
+        requestedNonMptNodes = requestedNonMptNodes - ref
+        requestedBlockBodies = requestedBlockBodies - ref
+        requestedReceipts = requestedReceipts - ref
```

Comment:
I'm wondering, should we add any request being handled by the terminated handler to our queues of pending requests? For example if a `FastSyncNodesRequestHandler` terminated we may have to add the nodes it was in charge of to `mptNodesQueue` and `nonMptNodesQueue` lists